src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -215,6 +217,11 @@ export class UserService {
     return user;
   }
 
+  async setModerator(user: User, moderator: Moderator): Promise<void> {
+    await this.userRepo.update(user.id, { usedRef: Config.moderators[moderator] });
+    await this.userDataService.updateUserDataInternal(user.userData, { moderator });
```

Comment:
Only set moderator ref, if the user doesn't have one?

Reply:
Hehe yeah true 😁